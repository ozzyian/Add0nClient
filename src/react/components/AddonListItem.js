import React from 'react';
const ipc = require('electron').ipcRenderer;
/**
 *
 */
class AddonListItem extends React.Component {
  /**
   *
   * @param {*} props
   */
  constructor(props) {
    super(props);
    this.updateAddon = this.updateAddon.bind(this);
    this.addon = this.props.addon;
    this.state = {
      updated: false,
      upateAvailable: true,
    };
  }

  /**
   *
   */
  componentDidMount() {
    ipc.on('update' + this.addon.id, () => {
      this.setState({updated: true});
    });
  }

  /**
   *
   */
  updateAddon() {
    ipc.send('update', this.addon);
  }
  /**
   * @return {*}
   */
  render() {
    if (this.state.updated) {
      return (
        <tr>
          <td>{this.props.addon.id}</td>
          <td>{this.props.addon.name}</td>
          <td>Updated</td>
          <td>{this.props.addon.gameVersionLatestFiles[0].projectFileName}</td>
          <td>{this.props.addon.authors[0].name}</td>
        </tr>
      );
    } else {
      return (
        <tr>
          <td>{this.props.addon.id}</td>
          <td>{this.props.addon.name}</td>
          <td>
            <button onClick={this.updateAddon}>Update</button>
          </td>
          <td>{this.props.addon.gameVersionLatestFiles[0].projectFileName}</td>
          <td>{this.props.addon.authors[0].name}</td>
        </tr>
      );
    }
  }
}

export default AddonListItem;
