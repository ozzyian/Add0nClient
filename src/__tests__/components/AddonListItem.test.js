import React from 'react';
import {shallow} from 'enzyme';
import AddonListItem from '../../react/components/AddonListItem';
import fs from 'fs';

describe('Snapshot tests', () => {
  let testAddon;
  beforeAll(() => {
    const testData = fs.readFileSync(
      './src/__tests__/resources/addon_info.json',
    );
    testAddon = JSON.parse(testData);
  });
  it('renders correctly', () => {
    const component = shallow(<AddonListItem addon={testAddon} />);
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when updating', () => {
    const component = shallow(<AddonListItem addon={testAddon} />);
    component.setState({updating: true, updateAvailable: false});
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when addon is updated', () => {
    const component = shallow(<AddonListItem addon={testAddon} />);
    component.setState({updating: false, updateAvailable: false});
    expect(component).toMatchSnapshot();
  });
});
