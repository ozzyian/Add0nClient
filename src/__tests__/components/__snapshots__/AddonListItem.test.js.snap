// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests renders correctly 1`] = `
<tr>
  <td>
    3358
  </td>
  <td>
    Deadly Boss Mods (DBM)
  </td>
  <td>
    <button
      onClick={[Function]}
    >
      Update
    </button>
  </td>
  <td>
    DBM-Core-1.13.57-54-gf85328d-classic.zip
  </td>
  <td>
    MysticalOS
  </td>
</tr>
`;

exports[`Snapshot tests renders correctly when addon is updated 1`] = `
<tr>
  <td>
    3358
  </td>
  <td>
    Deadly Boss Mods (DBM)
  </td>
  <td>
    Updated
  </td>
  <td>
    DBM-Core-1.13.57-54-gf85328d-classic.zip
  </td>
  <td>
    MysticalOS
  </td>
</tr>
`;

exports[`Snapshot tests renders correctly when updating 1`] = `
<tr>
  <td>
    3358
  </td>
  <td>
    Deadly Boss Mods (DBM)
  </td>
  <td>
    <Spinner
      animation="border"
      role="status"
    />
  </td>
  <td>
    DBM-Core-1.13.57-54-gf85328d-classic.zip
  </td>
  <td>
    MysticalOS
  </td>
</tr>
`;
